### PR TITLE
Libraries shims build fixes

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Delegate.cs [hunk 1]
```diff
@@ -52,7 +52,6 @@ public abstract partial class Delegate : ICloneable, ISerializable
         public static Delegate CreateDelegate(Type type, [DynamicallyAccessedMembers(AllMethods)] Type target, string method) => CreateDelegate(type, target, method, ignoreCase: false, throwOnBindFailure: true)!;
         public static Delegate CreateDelegate(Type type, [DynamicallyAccessedMembers(AllMethods)] Type target, string method, bool ignoreCase) => CreateDelegate(type, target, method, ignoreCase, throwOnBindFailure: true)!;
 
-#if !CORERT
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Delegate.cs:55]: Unnecessary ifdef

File: src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt [hunk 2]
```diff
@@ -95,8 +95,6 @@ add_library(System.IO.Compression.Native-static
     ${NATIVECOMPRESSION_SOURCES}
 )
 
-SET_TARGET_PROPERTIES(System.IO.Compression.Native-static PROPERTIES OUTPUT_NAME System.IO.Compression.Native)
```
Comment (stephentoub) [inline src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:98]: This just isn't needed anymore, or it's being set somewhere else, or it was just wrong?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt (hunk 2 above)
Comment (jkotas) [reply src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:98]: This project is building both static and dynamic libraries on Windows. Both of these builds were producing `System.IO.Compression.Native.lib`. Building dynamic .dll produces .dll and export .lib; building static library produces .lib. The real .lib and the export .lib were overwriting each other that leads to build race conditions. We have run into it in the runtimelab NativeAOT branch. Deleting this line makes the names distinct and fixes the build race condition.

Comment (stephentoub) [reply src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:98]: Got it.

**Pull request merged**